docs: new faq describing what can be made with SvelteKit (#10380)
sveltejs/kit: documentation/faq/10-what-can-i-create.md

---

> title: What kinds of applications can I make with SvelteKit?

Reviewer: ```suggestion
title: What can I make with SvelteKit?
```

---

> SvelteKit can be used to create several kinds of application with the support of [third-party tools](#integrations). Out of the box, SvelteKit supports many features including:

Reviewer: ```suggestion
SvelteKit can be used to create most kinds of applications with the support of [third-party tools](#integrations). Out of the box, SvelteKit supports many features including:
```

I am not the biggest fan of saying third-party tools but I think it gets the message across - maybe we can instead mention integrations after the bullet points?

> - Dynamic page content with [loaders](/docs/load) and [API endpoints](/docs/routing#server).

Reviewer: ```suggestion
- Dynamic page content with [loaders](/docs/load) and [API routes](/docs/routing#server).
```

- SEO-friendly dynamic content with [server-side rendering (SSR)](/docs/glossary#ssr).
- User-friendly progressively-enhanced interactive pages with SSR and [Form Actions](/docs/form-actions).

SvelteKit can also be deployed to a wide spectrum of hosted architectures via [adapters](/docs/adapters). In cases where SSR is used or server-side logic is added without using prerendering, those functions will be adapted to your configured back-end. Some examples include:

- Self-hosted dynamic web applications with a [Node.js backend](/docs/adapter-node).
- Serverless web applications with back-end loaders and APIs deployed as remote functions. See [zero-config deployments](/docs/adapter-auto) for popular deployment options.
- [Static pre-rendered sites](/docs/adapter-static) such as a blog or multi-page site hosted on a CDN or static host. Statically-generated sites are shipped without a backend.

> - [Single-page Applications (SPAs)](/docs/single-page-apps) with client-side routing and rendering for API-driven dynamic content. SPAs are shipped without a backend and are not server-rendered. This option is commonly chosen for simple servers such as those written in C, Golang, or Rust.

Reviewer: > This option is commonly chosen for simple servers such as those written in C, Golang, or Rust.

Not quite sure what you mean here?

Author: Non-Node.js servers won't be able to to server-render SvelteKit applications. Although this includes PHP, the languages that come up the most are Rust and Golang. I've included the C language for those writing UIs for embedded devices. 

Of course you can run a Node.js server alongside a Golang server, but this gets the point across that if the goal is to have a single Golang server, then the trade-off is to write an SPA. If this can be explained better let me know. At least with this inclusion, visitors will be able to type "Rust" or "Golang" in the search bar and they'll be led to this FAQ section.

Reviewer: Perhaps something like `This option is commonly chosen when serving SvelteKit from a custom server written in C, Golang, Rust, etc.`
